skimage/io/collection.py

```diff
@@ -72,20 +73,25 @@ def alphanumeric_key(s):
     return k
 
 
-class ImageCollection(object):
+def _is_multipattern(load_pattern):
+    """Helping function. Returns True if load_pattern contains a tuple, list,
+    or a string separated with os.pathsep."""
+    pattern = ((isinstance(load_pattern, str) and os.pathsep in load_pattern)
+               or (not isinstance(load_pattern, str) and
+                   isinstance(load_pattern, typing.Sequence) and
```

Comment:
I odn't think you should be checking against `typing.Sequence` rather `collections.abc.Sequence` https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence

Reply:
@jni can't remember why you suggested that, sorry. Could you back me up here? :smile:

Reply:
@alexandrejaguar probably ignorance. =P I think @hmaarrfk is right here. :joy:

Reply:
@hmaarrfk `typing.Sequence` says it's a generic version of `collections.abc.Sequence`. `typing.Sequence`:
```
This represents a generic version of type 'origin' with type arguments 'params'.
There are two kind of these aliases: user defined and special. The special ones
are wrappers around builtin collections and ABCs in collections.abc. These must
have 'name' always set. If 'inst' is False, then the alias can't be instantiated,
this is used by e.g. typing.List and typing.Dict.
```
I guess both of you are right, but I'll call `collections.abc.Sequence` anyway.

Reply:
Interesting. 

I guess after 6 months of trying to use the typing module I still can't really use it well. I'm going to wait until it settles more before bringing it into my personal work.

Reply:
Yeah, maybe we should stick to `collections.abc` right now.